test/dynamo/test_python_dispatcher.py

```diff
@@ -130,6 +135,184 @@ def fn(x, y):
         # No recompile
         self.assertEqual(counter.frame_count, 1)
 
+    def test_vmapped_autograd_function(self):
```

Comment:
For testing, we have a lot of autograd.Function we can test. A good comprehensive way to do this is to copy-paste the following and add torch.compile(backend="eager") testing to it https://github.com/pytorch/pytorch/blob/01554c7b5a224709ceffa4cbcdef6527bf77226a/test/functorch/test_vmap.py#L4341 (but only for the autograd_function_db).

The vmap tests are able to generate inputs with various in_dims, so that better exercises the logic. They're also able to generate vmap(vmap( tests.